ddtrace/tracer/option.go

```diff
@@ -1140,8 +1140,8 @@ func WithSpanID(id uint64) StartSpanOption {
 	}
 }
 
-// ChildOf tells StartSpan to use the given span context as a parent for the
-// created span.
+// ChildOf tells StartSpan to use the given span context as a parent for the created span.
+// ChildOf is deprecated, use span.StartChild instead.
```

Comment:
```suggestion
// ChildOf tells StartSpan to use the given span context as a parent for the created span.
//
// Deprecated: Use span.StartChild instead.
```

This syntax is supported in pkg.go.dev, showing a deprecated tag alongside the function, like in [ddtrace/tracer#SQLCommentInjectionMode](https://pkg.go.dev/gopkg.in/DataDog/dd-trace-go.v1@v1.58.0/ddtrace/tracer#SQLCommentInjectionMode):

![image](https://github.com/DataDog/dd-trace-go/assets/163009/5e6bf0ab-1e56-43b2-81b9-feffc83f7dab)